Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -350,16 +355,20 @@ extension InternalRouter where Self: Router {
     
     func announceImpendingReroute(at location: CLLocation) {
         delegate?.router(self, willRerouteFrom: location)
-        NotificationCenter.default.post(name: .routeControllerWillReroute, object: self, userInfo: [
-            RouteController.NotificationUserInfoKey.locationKey: location,
-        ])
+        
+        var userInfo: [RouteController.NotificationUserInfoKey: Any] = [
+            .locationKey: location,
+        ]
+        userInfo[.headingKey] = heading
+        
+        NotificationCenter.default.post(name: .routeControllerWillReroute, object: self, userInfo: userInfo)
     }
     
     func announce(reroute newRoute: Route, at location: CLLocation?, proactive: Bool) {
         var userInfo = [RouteController.NotificationUserInfoKey: Any]()
-        if let location = location {
-            userInfo[.locationKey] = location
-        }
+        // to do: WAS IT OKAY TO REMOVE THIS IF LET?
+        userInfo[.locationKey] = location
```

Comment:
Yes, that’s fine. If `location` is `nil`, it removes the `locationKey` entry from the dictionary. Sometimes that Swift behavior is surprising, but here the dictionary was empty to begin with.